File: 1_two_sum/two_sum.py
from typing import *


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = dict()
        for i, n in enumerate(nums):
            if target - n in hash_map:
                return [i, hash_map[target - n]]
            hash_map[n] = i

File: 1_two_sum/two_sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (hash.get(target - nums[i]) != null) {
                int arr[] = {i, hash.get(target - nums[i])};
                return arr;
            }
            hash.put(nums[i], i);
        }
        return null;
    }
}